api/actions/v1alpha1/k8sattributes_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/odigos-io/odigos/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type K8sLabelAttribute struct {
+	// The label name to be extracted from the pod.
+	// e.g. "app.kubernetes.io/name"
+	// +kubebuilder:validation:Required
+	LabelKey string `json:"labelKey"`
+	// The attribute key to be used for the resource attribute created from the label.
+	// e.g. "app.kubernetes.name"
+	// +kubebuilder:validation:Required
+	AttributeKey string `json:"attributeKey"`
+}
+
+type K8sAnnotationAttribute struct {
+	// The label name to be extracted from the pod.
+	// e.g. "kubectl.kubernetes.io/restartedAt"
+	// +kubebuilder:validation:Required
+	AnnotationKey string `json:"annotationKey"`
+	// The attribute key to be used for the resource attribute created from the label.
+	// e.g. "kubectl.kubernetes.restartedAte"
+	// +kubebuilder:validation:Required
+	AttributeKey string `json:"attributeKey"`
+}
+
+type K8sAttributesSpec struct {
+	ActionName string                       `json:"actionName,omitempty"`
+	Notes      string                       `json:"notes,omitempty"`
+	Disabled   bool                         `json:"disabled,omitempty"`
+	Signals    []common.ObservabilitySignal `json:"signals"`
+
+	// Collect the following container related attributes:
+	// k8s.container.name
+	// container.id
+	// container.image.name
+	// container.image.tag
+	CollectContainerAttributes bool `json:"collectContainerAttributes,omitempty"`
```

Comment:
Do we want empty semantics here?
```suggestion
	CollectContainerAttributes *bool `json:"collectContainerAttributes,omitempty"`
```